src/components/drop-down-options/drop-down-options.test.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { DropDownOption, DropDownOptions } from './drop-down-options';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { noop } from '../../helpers/func';
+import {
+  TEST_OPTIONS_SINGLE,
+  TEST_OPTIONS_SINGLE_VALUES,
+  TEST_OPTIONS_SMALL,
+  TEST_OPTIONS_SMALL_VALUES,
+} from './options.mock';
+
+describe('Button', () => {
+  it('should render correctly with default props', () => {
+    render(<DropDownOptions show={true} options={TEST_OPTIONS_SINGLE} onOptionSelect={noop} />);
+    expect(screen.queryByText(TEST_OPTIONS_SINGLE_VALUES[0])).toBeVisible();
+  });
+
+  it('should hide when component rerenders with show as false', () => {
+    const { rerender, unmount } = render(
+      <DropDownOptions show={true} options={TEST_OPTIONS_SINGLE} onOptionSelect={noop} />
+    );
+    expect(screen.queryByText(TEST_OPTIONS_SINGLE_VALUES[0])).toBeVisible();
+
+    unmount(); // i think this is needed since AnimatePresence keeps old component alive despite wait(...)
```

Comment:
Have you tried using something like this:

`await waitForElementToBeRemoved(() => screen.queryByText(TEST_OPTIONS_SINGLE_VALUES[0]));`

I know this usually works for child components of `<AnimatePresence>` but I'm not sure if this is something else.

Reply:
that was exactly it, thanks for the suggestion!